v2/call_option.go

```diff
@@ -119,6 +121,38 @@ func (r *boRetryer) Retry(err error) (time.Duration, bool) {
 	return 0, false
 }
 
+// OnHTTPCodes returns a Retryer that retries if and only if
+// the previous attempt returns a googleapi.Error whose status code is stored in
+// cc. Pause times between retries are specified by bo.
+//
+// bo is only used for its parameters; each Retryer has its own copy.
+func OnHTTPCodes(cc []int, bo Backoff) Retryer {
+	return &httpRetryer{
+		backoff: bo,
+		codes:   append([]int(nil), cc...),
+	}
+}
+
+type httpRetryer struct {
+	backoff Backoff
+	codes   []int
+}
+
+func (r *httpRetryer) Retry(err error) (time.Duration, bool) {
+	var gerr *googleapi.Error
```

Comment:
Side note: Maybe we should do the work to wrap these with APIError so gax only opperates on this type directly...

Reply:
I'm not sure what you mean, can you elaborate? The main sticking point for me is that `APIError` doesn't have a `Code` property (we'd need to decide on a transport-agnostic `Code` type or align on gRPC Codes).